File: android/app/src/main/java/com/soohoon/influe/data/model/ReelsNoteDTO.kt
package com.soohoon.influe.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

enum class ReelsNoteStatus(val rawValue: String, val displayName: String) {
    DRAFTING("drafting", "작성중"),
    READY_TO_UPLOAD("readyToUpload", "업로드 대기"),
    UPLOADED("uploaded", "업로드 완료");

    companion object {
        fun fromRawValue(raw: String): ReelsNoteStatus =
            entries.find { it.rawValue == raw } ?: DRAFTING
    }
}

@Serializable
data class ReelsNoteDTO(
    val id: String,
    @SerialName("workspace_id") val workspaceId: String,
    val title: String = "",
    @SerialName("attributed_content") val attributedContent: String? = null,
    @SerialName("plain_content") val plainContent: String = "",
    val status: String = "drafting",
    @SerialName("sponsorship_id") val sponsorshipId: String? = null,
    val tags: List<String> = emptyList(),
    @SerialName("is_pinned") val isPinned: Boolean = false,
    @SerialName("created_by") val createdBy: String? = null,
    @SerialName("created_at") val createdAt: String = "",
    @SerialName("updated_at") val updatedAt: String = ""
) {
    val reelsNoteStatus: ReelsNoteStatus
        get() = ReelsNoteStatus.fromRawValue(status)
}

@Serializable
data class ReelsNoteInsert(
    @SerialName("workspace_id") val workspaceId: String,
    @SerialName("created_by") val createdBy: String,
    val title: String = "",
    @SerialName("plain_content") val plainContent: String = "",
    @SerialName("attributed_content") val attributedContent: String? = null,
    val status: String = "drafting",
    val tags: List<String> = emptyList()
)

File: CreatorNote/Supabase/DTOs/ReelsNoteDTO.swift
import Foundation

struct ReelsNoteDTO: Codable, Identifiable, Sendable {
    let id: UUID
    var workspaceId: UUID
    var title: String
    var attributedContent: String?
    var plainContent: String
    var status: String
    var sponsorshipId: UUID?
    var tags: [String]
    var isPinned: Bool
    var createdBy: UUID?
    var updatedBy: UUID?
    var createdAt: Date
    var updatedAt: Date

    enum CodingKeys: String, CodingKey {
        case id, title, tags, status
        case workspaceId = "workspace_id"
        case attributedContent = "attributed_content"
        case plainContent = "plain_content"
        case sponsorshipId = "sponsorship_id"
        case isPinned = "is_pinned"
        case createdBy = "created_by"
        case updatedBy = "updated_by"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }

    var reelsNoteStatus: ReelsNoteStatus {
        ReelsNoteStatus(rawValue: status) ?? .drafting
    }
}
